test/bats/e2e-provider.bats

```diff
@@ -0,0 +1,353 @@
+}
+
+@test "CSI inline volume test with multiple secret provider class" {
+  result=$(kubectl exec secrets-store-inline-multiple-crd -- cat /mnt/secrets-store-0/$SECRET_NAME)
+  [[ "${result//$'\r'}" == "${SECRET_VALUE}" ]]
+
+  result=$(kubectl exec secrets-store-inline-multiple-crd -- cat /mnt/secrets-store-0/$KEY_NAME)
+  result_base64_encoded=$(echo "${result//$'\r'}" | base64 ${BASE64_FLAGS})
+  [[ "${result_base64_encoded}" == *"${KEY_VALUE_CONTAINS}"* ]]
+
+  result=$(kubectl exec secrets-store-inline-multiple-crd -- cat /mnt/secrets-store-1/$SECRET_NAME)
+  [[ "${result//$'\r'}" == "${SECRET_VALUE}" ]]
+
+  result=$(kubectl exec secrets-store-inline-multiple-crd -- cat /mnt/secrets-store-1/$KEY_NAME)
+  result_base64_encoded=$(echo "${result//$'\r'}" | base64 ${BASE64_FLAGS})
+  [[ "${result_base64_encoded}" == *"${KEY_VALUE_CONTAINS}"* ]]
+
+  result=$(kubectl get secret foosecret-0 -o jsonpath="{.data.username}" | base64 -d)
+  [[ "${result//$'\r'}" == "${SECRET_VALUE}" ]]
+
+  result=$(kubectl exec secrets-store-inline-multiple-crd -- printenv | grep SECRET_USERNAME_0) | awk -F"=" '{ print $2}'
+  [[ "${result//$'\r'}" == "${SECRET_VALUE}" ]]
+
+  run wait_for_process $WAIT_TIME $SLEEP_TIME "compare_owner_count foosecret-0 default 1"
+  assert_success
+
+  result=$(kubectl get secret foosecret-1 -o jsonpath="{.data.username}" | base64 -d)
+  [[ "${result//$'\r'}" == "${SECRET_VALUE}" ]]
+
+  result=$(kubectl exec secrets-store-inline-multiple-crd -- printenv | grep SECRET_USERNAME_1) | awk -F"=" '{ print $2}'
+  [[ "${result//$'\r'}" == "${SECRET_VALUE}" ]]
+
+  run wait_for_process $WAIT_TIME $SLEEP_TIME "compare_owner_count foosecret-1 default 1"
+  assert_success
+}
+
+@test "Test auto rotation of mount contents and K8s secrets - Create deployment" {
+  run kubectl create ns rotation
+  assert_success
+
+  envsubst < $BATS_TESTS_DIR/rotation/e2e_provider_synck8s_v1alpha1_secretproviderclass.yaml | kubectl apply -n rotation -f -
+  envsubst < $BATS_TESTS_DIR/rotation/pod-synck8s-e2e-provider.yaml | kubectl apply -n rotation -f -
+
+  kubectl wait -n rotation --for=condition=Ready --timeout=60s pod/secrets-store-inline-rotation
+
+  run kubectl get pod/secrets-store-inline-rotation -n rotation
+  assert_success
+}
+
+@test "Test auto rotation of mount contents and K8s secrets" {
+  result=$(kubectl exec -n rotation secrets-store-inline-rotation -- cat /mnt/secrets-store/$SECRET_NAME)
+  [[ "${result//$'\r'}" == "secret" ]]
+
+  result=$(kubectl get secret -n rotation rotationsecret -o jsonpath="{.data.username}" | base64 -d)
+  [[ "${result//$'\r'}" == "secret" ]]
+  
+  sleep 60
+
+  result=$(kubectl exec -n rotation secrets-store-inline-rotation -- cat /mnt/secrets-store/$SECRET_NAME)
+  [[ "${result//$'\r'}" == "rotated" ]]
+
+  result=$(kubectl get secret -n rotation rotationsecret -o jsonpath="{.data.username}" | base64 -d)
+  [[ "${result//$'\r'}" == "rotated" ]]
+}
+
+@test "Test filtered-watch-secret=false" {
```

Comment:
We can remove this test as filtered watch secret is enabled by default and `--filtered-watch-secret` is deprecated.

Reply:
done